federation/cmd/federated-controller-manager/app/options/options.go

```diff
@@ -0,0 +1,95 @@
+	// allowed to sync concurrently. Larger number = more responsive replica
+	// management, but more CPU (and network) load.
+	ConcurrentSubRCSyncs int `json:"concurrentSubRCSyncs"`
+	// clusterMonitorPeriod is the period for syncing ClusterStatus in cluster controller.
+	ClusterMonitorPeriod unversioned.Duration `json:"clusterMonitorPeriod"`
+	// federatedAPIQPS is the QPS to use while talking with ubernetes apiserver.
+	FederatedAPIQPS float32 `json:"federatedAPIQPS"`
+	// federatedAPIBurst is the burst to use while talking with ubernetes apiserver.
+	FederatedAPIBurst int `json:"federatedAPIBurst"`
+	// enableProfiling enables profiling via web interface host:port/debug/pprof/
+	EnableProfiling bool `json:"enableProfiling"`
+	// leaderElection defines the configuration of leader election client.
+	LeaderElection componentconfig.LeaderElectionConfiguration `json:"leaderElection"`
+}
+
+// CMServer is the main context object for the controller manager.
+type CMServer struct {
+	ControllerManagerConfiguration
+	Master          string
+	ApiServerconfig string
+}
+
+const (
+	// FederatedControllerManagerPort is the default port for the ubernetes controller manager status server.
+	// May be overridden by a flag at startup.
+	FederatedControllerManagerPort = 10252
+)
+
+// NewCMServer creates a new CMServer with a default config.
+func NewCMServer() *CMServer {
+	s := CMServer{
+		ControllerManagerConfiguration: ControllerManagerConfiguration{
+			Port:                 FederatedControllerManagerPort,
+			Address:              "0.0.0.0",
+			ConcurrentSubRCSyncs: 5,
+			ClusterMonitorPeriod: unversioned.Duration{40 * time.Second},
+			FederatedAPIQPS:      20.0,
+			FederatedAPIBurst:    30,
+			LeaderElection:       leaderelection.DefaultLeaderElectionConfiguration(),
+		},
+	}
+	return &s
+}
+
+// AddFlags adds flags for a specific CMServer to the specified FlagSet
+func (s *CMServer) AddFlags(fs *pflag.FlagSet) {
+	fs.IntVar(&s.Port, "port", s.Port, "The port that the controller-manager's http service runs on")
+	fs.Var(componentconfig.IPVar{&s.Address}, "address", "The IP address to serve on (set to 0.0.0.0 for all interfaces)")
+	fs.IntVar(&s.ConcurrentSubRCSyncs, "concurrent-subRc-syncs", s.ConcurrentSubRCSyncs, "The number of subRC syncing operations that will be done concurrently. Larger number = faster endpoint updating, but more CPU (and network) load")
+
+	fs.DurationVar(&s.ClusterMonitorPeriod.Duration, "cluster-monitor-period", s.ClusterMonitorPeriod.Duration, "The period for syncing ClusterStatus in ClusterController.")
+	fs.BoolVar(&s.EnableProfiling, "profiling", true, "Enable profiling via web interface host:port/debug/pprof/")
+	fs.StringVar(&s.Master, "master", s.Master, "The address of the Kubernetes API server (overrides any value in kubeconfig)")
+	fs.StringVar(&s.ApiServerconfig, "federatedconfig", s.ApiServerconfig, "Path to ApiServerconfig file with authorization and master location information.")
```

Comment:
We can keep the kubeconfig name as in kube api server.
We will use the same kubeconfig file